### DISCUSSION TITLE
Cash Drawer Support

### DISCUSSION THREAD

**Discussion by rsidhaarth** [Blair2004/NexoPOS#383] [Ideas]
Hi Team,

We use cash drawer and I believe most people do. It will be much better if you add support to trigger cash drawer only for cash payments. And also allow people to open cash drawer manually using pin during other times (those who have access to cash drawer and with pin)

**Comment by Blair2004** [comment]
Mainly cash drawer support implies being able to open the cash drawer :
- After checkout
- Manually
Right ?

**Reply by rsidhaarth** [reply]
1) It needs to open automatically if the mode of payment is cash or cash + card
2) Sometimes, the cashier needs to access the drawer manually. In that case, we need an option to choose "Open Cash Drawer" which will show a prompt to enter a PIN. Those who have access to the cash drawer can enter their PIN and it will get opened. And all the manual action needs to be logged.

**Comment by rsidhaarth** [comment]
After testing the POS I found out Cash Register is something different. So we need an option called Cash Drawer, where only cash payment goes in and goes out as Cash Register.

But we need to have opening cash and then closing cash, if there is a mismatch in the closing cash, then the cashier needs to provide a comment for the same.

Closing Cash = Opening Cash + Sales (Cash Payment)

So Cash Drawer needs to be a part of the Cash Register. Cash Register will show all the incoming and outgoing transactions. Cash Drawer will show only cash related incoming and outgoing transactions.

**Reply by Blair2004** [reply]
Do you mean the actual system doesn't have opening and closing cash ?
The Cash Register usually has a cash drawer. From the cash register, it can perform disbursement.
We're about to create a guide on this.

**Reply by rsidhaarth** [reply]
There is an issue with the cash register. Let me explain the scenario.

1) I opened the cash register as 0rs
2) We got sales worth 1000rs
3) I closed the cash register by mentioning closing cash as 1000rs

Then I saw that closing cash is entered in the Expense. Not sure why? I believe it is asking closing cash to verify the total and if there is a discrepancy then it has to ask the reason and then only mark the difference in expense and not the whole amount.

Now the cash register is for all transactions (cash, card,bank).

But POS need cash drawer option, because most of the issue happens with cash transaction. And so store needs to have a option to use 

1) Cash Drawer
2) Cash Register
3) Both

If cash drawer, then they have to enter the opening cash (only cash amount in the drawer) and then you will be making entry to cash drawer for the all-cash transaction as cash in. And then during the closing, you have to show the closing amount (total sales - cashout ) and then the cashier will enter the closing amount. If there is a difference then they will enter the reason and close it. And the difference will be entered in expense as a discrepancy.

Also you can provide an option to verify the cash register as well. You can plan for this.
